crates/shared/src/event_handling.rs

```diff
@@ -23,7 +28,7 @@ where
     block_retriever: B,
     contract: C,
     store: S,
-    last_handled_block: Option<u64>,
+    last_handled_blocks: Vec<BlockNumberHash>,
```

Comment:
Would be useful to have a short comment here explaining the semantics of this field. Or maybe a comment on the struct outlining the algorithm.

Reply:
Added algorithm description.